### PR TITLE
docs: update the README.md file with detailed instructions

### LINKED ISSUE #4: Create README for this template
Make sure we don't forget to include:

1. How to install it.
1. How to use it.
1. The params it supports.
1. The requirement of `operationId` on each operation (otherwise, the code would not work).
1. This README should serve as a template for the rest of the generator templates.

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**

- updated README.md file.

**Related issue(s)**
Resolves #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,82 @@
...
+##
+
+# Go to the generated server
+cd output
+
+# Build generated application
+npm i
+
+# Start server
+npm start
```
Comment (derberg) [inline README.md:59]: consider adding here instructions about installing mqtt client with proper description that it is just to quickly see the template in action

```bash
npm install mqtt -g
mqtt pub -t 'light/measured' -h 'test.mosquitto.org' -m '{"id": 1, "lumens": 3, "sentAt": "2017-06-07T12:34:32.000Z"}'
```

File: README.md [hunk 2]
```diff
@@ -1,24 +1,82 @@
...
 |Name|Description|Required|Example|
 |---|---|---|---|
 |server|The server you want to use in the code.|Yes|`production`|
 
-## Supported protocols
+## Custom hooks that you can disable
+
+The functionality of this template is extended with different hooks that you can disable like this in the Generator CLI: `-d HOOK_TYPE1=HOOK_NAME1,HOOK_NAME2 -d HOOK_TYPE2`
 
 * AMQP
```
Comment (derberg) [inline README.md:81]: these protocols should not be mentioned here

File: README.md [hunk 3]
```diff
@@ -1,24 +1,82 @@
...
+You can configure this template by passing different parameters in the Generator CLI: `-p PARAM1_NAME=PARAM1_VALUE -p PARAM2_NAME=PARAM2_VALUE`
 
 |Name|Description|Required|Example|
 |---|---|---|---|
 |server|The server you want to use in the code.|Yes|`production`|
 
-## Supported protocols
+## Custom hooks that you can disable
+
+The functionality of this template is extended with different hooks that you can disable like this in the Generator CLI: `-d HOOK_TYPE1=HOOK_NAME1,HOOK_NAME2 -d HOOK_TYPE2`
```
Comment (derberg) [inline README.md:79]: there is just one hook used in this template. This template has no specific hooks as there is no hooks dir, but it is using a generic hook -> https://github.com/asyncapi/nodejs-template/blob/master/package.json#L91 please document it

**Review by derberg — COMMENTED** [review]
looks good so far, added few comments.

Please also add other sections from https://github.com/asyncapi/nodejs-ws-template and please make sure we have the same order of sections there.

Would be great to add the same dependency for markdow toc generation and have an npm script for it

**Comment by KhudaDad414** [comment]
thanks. I've started working on this.

**Comment by KhudaDad414** [comment]
@derberg Please check it again. I think I solved all of the issues.

File: README.md [hunk 4]
```diff
@@ -1,26 +1,160 @@
...
+* [Kafka](https://en.wikipedia.org/wiki/Apache_Kafka)
+* [WebSocket](https://en.wikipedia.org/wiki/WebSocket)
+
+
+
+## How to use the template
+
+This template must be used with the AsyncAPI Generator. You can find all available options [here](https://github.com/asyncapi/generator/).
+
+Note: You can find a complete tutorial on AsyncAPI generator using this template [here](https://www.asyncapi.com/docs/tutorials/streetlights).
```
Comment (derberg) [inline README.md:59]:
```suggestion
> You can find a complete tutorial on AsyncAPI Generator using this template [here](https://www.asyncapi.com/docs/tutorials/streetlights). 
```

**Review by derberg — COMMENTED** [review]
It looks great man, just added one minor comment, and please also add https://github.com/asyncapi/nodejs-ws-template/blob/master/.all-contributorsrc so we can use allcontributors bot, to for example add you after we merge this PR

**Comment by derberg** [comment]
And please update to latest master 🙏🏼

**Comment by KhudaDad414** [comment]
@derberg made the changes that you asked for.

**Review by derberg — APPROVED** [review]
@KhudaDad414 Thank you and thanks for another fix that enables https://bit.ly/asyncapi to work with this template without any modifications

**Comment by derberg** [comment]
@all-contributors please add @KhudaDad414 for docs, code

**Comment by derberg** [comment]
@all-contributors please add @derberg for code

**Pull request merged**